tests/test_reader_api.py

```diff
@@ -1,105 +1,281 @@
+            {"name": "my_float", "type": "float64", "type_category": "float"},
+            {"name": "my_bool", "type": "bool_", "type_category": "boolean"},
+            {"name": "my_nullable_bool", "type": "bool_", "type_category": "boolean"},
+            {"name": "my_date", "type": "date32", "type_category": "timestamp"},
+            {
+                "name": "my_datetime",
+                "type": "timestamp(s)",
+                "type_category": "timestamp",
+            },
+            {"name": "my_int", "type": "int64", "type_category": "integer"},
+            {"name": "my_string", "type": "string", "type_category": "string"},
+        ]
+    }
 
 
 @pytest.mark.parametrize("data_format", ["jsonl", "csv"])
 def test_inferred_cols_pandas_types(data_format):
     df = reader.read(f"tests/data/all_types.{data_format}")
     test = df.dtypes.to_dict()
+
     assert isinstance(test["i"], pd.core.arrays.integer.Int64Dtype)
     assert isinstance(test["my_float"], type(np.dtype("float64")))
     assert isinstance(test["my_bool"], pd.core.arrays.boolean.BooleanDtype)
+    assert isinstance(test["my_string"], pd.core.arrays.string_.StringDtype)
+
     if data_format == "jsonl":
         pytest.skip("Pandas cannot infer bool with nulls from JSON datasets")
     else:
         assert isinstance(test["my_nullable_bool"], pd.core.arrays.boolean.BooleanDtype)
 
-    assert isinstance(test["my_string"], pd.core.arrays.string_.StringDtype)
+
+@pytest.mark.parametrize("data_format, expected_class", test_default_file_types_reader)
+class Test_get_default_reader:
+    def test_get_default_reader_type_from_file_format(
+        self, data_format, expected_class
+    ):
+        """
+        Test that get_reader_for_file_format retrieves the correct reader for
+        each file format.
+        """
+        actual = get_reader_for_file_format(data_format, None)
+        assert isinstance(actual, expected_class)
+
+    @pytest.mark.parametrize("use_meta", [True, False])
+    def test_infer_default_reader_from_file_path(
+        self,
+        monkeypatch: pytest.MonkeyPatch,
+        data_format,
+        expected_class,
+        use_meta,
+        test_meta,
+    ):
+        """
+        Test that get_reader_for_file_format can infer the file type from the file name
+        and then retrieve the correct reader for each file format, when no engine
+        argument is supplied.
+        """
+        if use_meta:
+            meta = test_meta
+        else:
+            meta = None
+
+        with tempfile.NamedTemporaryFile(suffix="." + data_format) as temp_file_name:
+            temp_out_file = temp_file_name.name
+
+        # stub out reader.expected_class.read for True (rather than pd.DataFrame)
+        # so actual --> True if the expected_class is called
+        monkeypatch.setattr(expected_class, "read", lambda *args, **kwargs: "stub")
+
+        actual = reader.read(input_path=temp_out_file, metadata=meta)
+
+        assert actual == "stub"
```

Comment:
Why "stub"?

Reply:
Good spot. This test and the later test should revert back to `monkeypatch.setattr(,, lambda *args, **kwargs: True)` and `assert is True`. I thought I was having linting issues with my assertion, but I can't replicate now.


For info:
"stub" matches my understanding of the [mocking language in this article](https://martinfowler.com/articles/mocksArentStubs.html#TheDifferenceBetweenMocksAndStubs): 

> Stubs provide canned answers to calls made during the test, usually not responding at all to anything outside what's programmed in for the test.

Reply:
I am unsure if the return `monkeypatch.setattr(,, lambda *args, **kwargs: True)` would work in this one as `reader.read` returns a dataframe, and I imagine that evaluates to True?

Reply:
It appears to work to evaluate `is True`, but that is the error I was trying to avoid with "stub"

"`== True`" fails linting, as True should be evaluated with `is`.

I'm wondering if leaving it as "stub" works out to be more explicit here

Reply:
No strong feelings either way - leaving as is